src/module-elasticsuite-catalog/etc/di.xml

```diff
@@ -330,4 +330,41 @@
     <type name="\Magento\Catalog\Api\Data\ProductAttributeInterface">
         <plugin name="resetIndexConfigAfterAttributeSave" type="\Smile\ElasticsuiteCatalog\Plugin\Catalog\Eav\AttributePlugin" />
     </type>
+
+    <type name="Smile\ElasticsuiteCore\Model\Search\Request\Field\Mapper">
+        <arguments>
+            <argument name="fieldNameMapping" xsi:type="array">
+                <item name="price" xsi:type="string">price.price</item>
+                <item name="position" xsi:type="string">category.position</item>
+                <item name="category_ids" xsi:type="string">category.category_id</item>
+            </argument>
+        </arguments>
+    </type>
+
+    <type name="Smile\ElasticsuiteCatalogRule\Model\Rule\Condition\Product\AttributeList">
+        <arguments>
+            <argument name="requestFieldMapper" xsi:type="object">Smile\ElasticsuiteCatalog\Model\Layer\Filter\Field\Mapper</argument>
+        </arguments>
+    </type>
+    <virtualType name="Smile\ElasticsuiteCatalog\Model\Layer\Filter\Field\Mapper" type="Smile\ElasticsuiteCore\Model\Search\Request\Field\Mapper">
+        <arguments>
+            <argument name="fieldNameMapping" xsi:type="array">
+                <item name="price" xsi:type="string">price.price</item>
```

Comment:
I'm not sure the Mapper should vary as a virtual type. A mapping is something "consistant", it can be a bit blurry that in some contexts the mapping for "price" could  be different.

And about "categories" mapping, I'm not sure it should "appear" only in particular contexts.

Reply:
Virtual type has been removed.